### ISSUE TITLE
Add support for `rsa-sha2-256` algorithm for signing ssh connection

### ISSUE THREAD

**Issue by cfouche3005** [quexten/goldwarden#131]
When using goldwarden for ssh login with a RSA key, I get this error in the ssh connection :
```
debug3: sign_and_send_pubkey: using publickey-hostbound-v00@openssh.com with RSA SHA256:XXX
debug3: sign_and_send_pubkey: signing using rsa-sha2-256 SHA256:XXX
agent key RSA SHA256:XXX returned incorrect signature type
sign_and_send_pubkey: no mutual signature supported
```
After doing some research, I have found this error appear when the SSH agent implementation doesn't support the signing algorithm.
As OpenSSH [8.2](https://www.openssh.com/txt/release-8.2) supported signing algorithm are 
- rsa-sha2-256 (tested, doesn't work with goldwarden-ssh-agent)
- rsa-sha2-512 (not tested)
- ssh-ed25519 (tested, work with goldwarden-ssh-agent)
- ecdsa-sha2-nistp256 (not tested)
- ecdsa-sha2-nistp384 (not tested)
- ecdsa-sha2-nistp521(tested, work with goldwarden-ssh-agent)

Is it possible too add support for rsa-sha2-256 and rsa-sha2-512 (and possibly other algorithm) ?

**Comment by csutcliff** [comment]
This should be covered by #123, just needs a release.

**Comment by quexten** [comment]
https://github.com/quexten/goldwarden/pull/123

@csutcliff implemented the extended agent in the PR (currently only on main branch, not release). I think this should fix rsa-sha2-256/rsa-sha2-512 on the server side?

For the keys stored in the vault, it's currently only ed25519 keys since importing is not implemented yet anyways. Open to PRs for RSA generation / import, otherwise this will take a while.

**Comment by cfouche3005** [comment]
> For the keys stored in the vault, it's currently only ed25519 keys since importing is not implemented yet anyways. Open to PRs for RSA generation / import, otherwise this will take a while.

RSA/ECDSA work perfectly with the vault (minus the signing problem) thanks to the OpenSSH key conversion

**Comment by quexten** [comment]
Is included in 0.2.14/15. Please re-open if anything does not work as expected.
